modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -5126,6 +5126,12 @@ public void onNodeLeft(final ClusterNode node) {
 
                             if (crd0 == null)
                                 finishState = new FinishState(null, initialVersion(), null);
+
+                            if (dynamicCacheStartExchange() &&
+                                exchActions.cacheStartRequiredAliveNodes().contains(node.id())) {
+                                exchangeGlobalExceptions.put(cctx.localNodeId(), new ClusterTopologyCheckedException(
```

Comment:
I think we should also set the `exchangeLocE` param.
Let's add a dedicated test for starting caches with the required alive nodes which is not related to the snapshot procedure.

Reply:
added DynamicCacheStartFailsOnNodeLeftTest